### DISCUSSION TITLE
ARM SMMU parameter values

### DISCUSSION THREAD

**Discussion by sammy17** [gem5/gem5#1124] [Q&A] [answered]
I just wanted to ask whether there is any reasoning behind the parameter values specified in ARM SMMU: https://github.com/gem5/gem5/blob/stable/src/dev/arm/SMMUv3.py

For example, why request port size is set to 16 bytes? Is this representative of the real ARM SMMUv3 hardware? 

Similarly, what is the reasoning behind setting the xlate_slots value of the device interface to 16 while the xlate_slots of the SMMU is set to 64? 

I would really appreciate any help in understanding this.

**Comment by giactra** [comment, marked as answer]
> I just wanted to ask whether there is any reasoning behind the parameter values specified in ARM SMMU: https://github.com/gem5/gem5/blob/stable/src/dev/arm/SMMUv3.py

> For example, why request port size is set to 16 bytes? Is this representative of the real ARM SMMUv3 hardware?

Hi @sammy17, as I said in one of our earlier discussions, the SMMUv3 in gem5 is an architectural model and does not claim to be representative of any real SMMUv3 implementation.

The parameters (latencies and buffer sizes) allow an user to experiment with them, but no importance shall be given to their default value.

> Similarly, what is the reasoning behind setting the xlate_slots value of the device interface to 16 while the xlate_slots of the SMMU is set to 64?

Same as above. Wrt this I can only add it makes sense for the SMMU xlate_slots to be a multiple of the device interface one in a system with multiple devices(interfaces).

**Reply by sammy17** [reply]
I see, thank you very much for the help @giactra !!
